### ISSUE TITLE
Indirect access to mailbox (according to documentation

### ISSUE THREAD

**Issue by JohWayne** [AbuseIO/AbuseIO#280]
Hi

I have a problem of understanding and hope that someone can help. According to the [documentation](https://docs.abuse.io/en/latest/data_collection/#how-incoming-data-is-received) it should be possible to receive mails instead of your own MTA by accessing a mailbox (IMAP or POP3?) and parse them.

I just don't understand how I can make this happen. There are no instructions for this, I have overlooked or misunderstood this feature? I'm using AbuseIO 4.0.3.

Can someone help me with that? How can I give AbuseIO access to an existing mailbox (which has been set up on the mail server for this purpose)? Which file (module) is responsible for this und wie kann ich dieses aktivieren?

Thank you in advance.

PS: What does this option/URL mean (as it is not available in my installation)?

`'ash' => [
        'url'                               => 'https://abuseio.isp.local/ash/',
    ],`

Best regards,
Joh

**Comment by kruisdraad** [comment]
Also,

The ASH->URL is a URL you have (each abuseio has it) which should be set to your hostname. if if you access the admin interface using abuseio.joh.de, then the default ASH url would be abuseio.joh.de/ash/

i know there are examples of mailbox usage, but i don't have one myself. @joverthegrey  @marknl perhaps know.

**Comment by JohWayne** [comment]
Thanks for your explanation, @kruisdraad. As it seems, this URL doesn't work for me, despite the modification in the config file (main.php).  I'm getting a 404 error:

`Status: HTTP/1.0 404 Not Found`
> Sorry, the page you are looking for could not be found.

I'll have to look into that. Maybe the logfiles will give me more information about it

However, a solution for accessing the mail account would be exciting. I hope someone can help me with that. I am working on a test environment for the evaluation of AbuseIO as abusedesk in our company, and hope to get this working.

**Comment by kruisdraad** [comment]
https://abuseio.isp.local/ash/ = a prefix, it will open something like https://abuseio.isp.local/ash/ticket/token/etc which does work. In your ADMIN GUI with you open a ticket you see the full ASH url. The first is used then /ticket/token/etc is added.

**Comment by joverthegrey** [comment]
Hi @JohWayne,
AbuseIO itself cannot POP or IMAP a mailbox, you'll need to use an external tool like fetchmail / procmail to retrieve the mail from the mailbox and serve it to AbuseIO.

The abuseio-docker has the config files for both, you probably want to edit them depending on  your situation.

https://github.com/AbuseIO/abuseio-docker/blob/master/README.md
https://github.com/AbuseIO/abuseio-docker/blob/master/config/fetchmail/fetchmailrc
https://github.com/AbuseIO/abuseio-docker/blob/master/config/procmail/procmailrc

**Comment by JohWayne** [comment]
Thank you very much for your input, @joverthegrey. That helped me in part. Fetchmail works so far, mails are retrieved and stored locally under /home/abuseio/mail.

Which I'm not quite sure about yet. Do I have to tell AbuseIO to read these mails from there or does this already happen automatically? Does AbuseIO automatically check if mails have arrived locally for its user? I don't quite understand the concept behind it yet...

**Comment by joverthegrey** [comment]
Fetchmail pulls the mails to a (local) user and then you can tell procmail as that user to feed them to abuseio

'''
PATH=$HOME/bin:/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin
MAILDIR=/opt/abuseio
LOGFILE=/log/abuseio/procmail.log
VERBOSE=yes

:0
|php artisan email:receive
'''

in this case procmail checks the users .procmailrc (in /home/abuseio/.procmailrc) and sees that it need to pipe all mails (when they arrive) to the command 'php artisan email:receive' which will put the mails in AbuseIO's queue. 

So in short: Fetchmail retrieves the mail from the external mailbox and delivers it to the Procmail, procmail feeds it to abuseio.

It is also possible to leave Procmail out of this chain and use '/usr/bin/php <abuseiodir>/artisan email:receive' as mda command in the fetchmailrc

With procmail you can do some more stuff with the mails (like save them to an folder, do some auto-replying etc). 

Hope that this will help :)

**Comment by JohWayne** [comment]
I would like to thank you for your support.

With a few adjustments it worked out in the end. Now the testing can start. ;)
